src/wr_logic_ai/nodes/state_machine/state_machine.py

```diff
@@ -1,95 +1,127 @@
+        self.mux_long_range.state = NavigationState.NAVIGATION_STATE_LONG_RANGE
+        self.mux_short_range = NavigationState()
+        self.mux_short_range.state = NavigationState.NAVIGATION_STATE_SHORT_RANGE
 
-
-    def process_event(self, data: NavigationStateMsg):
-        print(data.nav_state_type)
-        if data.nav_state_type == NavigationStateMsg.NAV_STATE_TYPE_SUCCESS:
-            self.evSuccess()
-        elif data.nav_state_type == NavigationStateMsg.NAV_STATE_TYPE_ERROR:
-            self.evError()
-        else:
-            raise TypeError
-    
     def on_enter_stInit(self) -> None:
         print("\non enter stInit")
         self._mgr.read_coordinates_file()
         self.evUnconditional()
 
     def on_exit_stInit(self) -> None:
-        self._mgr.next_line()
+        if (self._mgr.next_line()):
+            self.evComplete
 
-    def on_enter_stLongRange(self)->None:
+    def on_enter_stLongRange(self) -> None:
         print("\non enter stLongRange")
-        # self.timer = rospy.Timer(rospy.Duration(0.2), self.publish)
-        #self.pub_nav = rospy.Publisher('/target_coord', TargetMsg, queue_size=1)
-        print(self._mgr.get_coordinate())
-        #rospy.spin()
+        self.timer = rospy.Timer(rospy.Duration(0.2), lambda _: self.mux_pub.publish(self.mux_long_range)) 
+        client = actionlib.SimpleActionClient("LongRangeActionServer", LongRangeAction)
+        client.wait_for_server()
+        goal = LongRangeGoal(target_lat = self._mgr.get_coordinate()["lat"], target_long = self._mgr.get_coordinate()["long"])
+        client.send_goal(goal)
+        client.wait_for_result()
+        as_state = client.get_state()
+        if as_state == GoalStatus.SUCCEEDED:
+            self.evSuccess()
+        elif as_state == GoalStatus.ABORTED:
+            self.evError()
 
     def on_exit_stLongRange(self) -> None:
-        #self.timer.shutdown()
-        pass
+        print("Exting Long Range")
+        self.timer.shutdown()
 
     def on_enter_stLongRangeRecovery(self) -> None:
         print("\non enter stLongRangeRecovery")
         if self._mgr is None:
             raise ValueError
         else:
             self._mgr.previous_line()
-            print(self._mgr.get_coordinate())
-            #self.timer = rospy.Timer(rospy.Duration(0.2), self.publish)
-            #self.pub_nav = rospy.Publisher('/target_coord', TargetMsg, queue_size=1)
-            #rospy.spin()
+            print(self._mgr.get_coordinate())  
+            self.timer = rospy.Timer(rospy.Duration(0.2), lambda _: self.mux_pub.publish(self.mux_long_range))           
+            client = actionlib.SimpleActionClient("LongRangeActionServer", LongRangeAction)
+            client.wait_for_server()
+            goal = LongRangeGoal(target_lat = self._mgr.get_coordinate()["lat"], target_long = self._mgr.get_coordinate()["long"])
+            client.send_goal(goal)
+            client.wait_for_result()
+            as_state = client.get_state()
```

Comment:
Use `send_goal_and_wait` (https://docs.ros.org/en/diamondback/api/actionlib/html/classactionlib_1_1simple__action__client_1_1SimpleActionClient.html#a3de1204015fe23bfd1b47b7fe164df34)

Reply:
Addressed in [277b00a](https://github.com/WisconsinRobotics/WRover_Software/pull/48/commits/277b00a47c8537aba6ba2aeeb6c09d9b5ea4d3eb).